types/kafka-node/index.d.ts

```diff
@@ -6,26 +6,27 @@
 // # Classes
 export class Client {
     constructor(connectionString: string, clientId?: string, options?: ZKOptions, noBatchOptions?: AckBatchOptions, sslOptions?: any);
-    close(callback?: () => void): void;
-    topicExists(topics: string[], callback: (err?: TopicsNotExistError | any) => any): void;
+    close(cb?: () => void): void;
+    topicExists(topics: string[], cb: (error?: TopicsNotExistError | any) => any): void;
     refreshMetadata(topics: string[], cb?: (error?: any) => any): void;
-    sendOffsetCommitV2Request(group: string, generationId: number, memberId: string, commits: OffsetCommitRequest[], cb: () => void): void;
+    sendOffsetCommitV2Request(group: string, generationId: number, memberId: string, commits: OffsetCommitRequest[], cb: (error: any, data: any) => any): void;
 }
 
 export class KafkaClient extends Client {
     constructor(options?: KafkaClientOptions);
+    on(eventName: "ready", cb: () => any): this;
+    on(eventName: "error", cb: (error: any) => any): this;
     connect(): void;
 }
 
 export class Producer {
-    constructor(client: Client, options?: ProducerOptions, customPartitioner?: any);
+    constructor(client: Client, options?: ProducerOptions, customPartitioner?: CustomPartitioner);
     on(eventName: "ready", cb: () => any): void;
     on(eventName: "error", cb: (error: any) => any): void;
     send(payloads: ProduceRequest[], cb: (error: any, data: any) => any): void;
-    createTopics(topics: string[], async: true, cb: (error?: any, data?: any) => any): void;
-    createTopics(topics: string[], async: false, cb: () => any): void;
-    createTopics(topics: string[], cb: (error?: any, data?: any) => any): void;
-    close(): void;
+    createTopics(topics: string[], async: boolean, cb: (error: any, data: any) => any): void;
```

Comment:
```(error: any, data: any) => any)```
is the same as
```(error?: any, data?: any) => any)```
(those "?"s are useless as ```any``` covers ```undefined```)
and it also therefore covers
```() => any)```

So this single declaration seems equivalent to (and less "wordy" than) the three it replaced.